afew/MailMover.py

```diff
@@ -72,31 +75,45 @@ def move(self, maildir, rules):
                     if self.dry_run:
                         continue
                     try:
-                        shutil.copy2(fname, self.get_new_name(fname, destination))
-                        to_delete_fnames.append(fname)
-                    except shutil.SameFileError:
-                        logging.warn("trying to move '{}' onto itself".format(fname))
-                        continue
+                        logging.info("Copy original mail: "+fname)
+                        logging.info("Path of mail: "+os.path.dirname(fname))
+                        logging.info("Copy to: "+destination)
+                        dbpath = self.db.get_path()
+                        logging.info("Path of DB: "+dbpath)
+                        relpath=os.path.relpath(os.path.dirname(destination), dbpath)
+                        filename = os.path.basename(fname)
+                        logging.info("filename: "+filename)
+                        logging.info("Relative path DB -> file: " + relpath)
+                        # 1 COPY
+                        shutil.copy2(fname, self.get_new_name(
+                                                            fname,
+                                                            destination))
+                        dest_file_rel = dbpath+'/'+relpath+'/'+filename
+                        logging.info("Add new copy to the DB: "+dest_file_rel)
+                        # OPEN DB R/W
+                        logging.info("Open DB R/W")
+                        # print vars(self.db)
+                        # 2 Add NEW MESSAGE TO DB
+                        moved_message = notmuch.Database.add_message(self.db, dest_file_rel)
+                        logging.info("Delete original mail file: "+fname)
+                        os.remove(fname)
+                        # 3 REMOVE ORIGINAL
+                        logging.info("Delete the original mail in DB")
+                        # 4 REMOVE ORIGINAL FROM DB
+                        notmuch.Database.remove_message(self.db, fname)
```

Comment:
While I do favor some logging, this whole block seems to be wayy to extensive.

Most of the statements are merely printing out variable values. Maybe one or two lines with loglevel debug (something like `{} -> {}`.format(oldpath, newpath)`) should be enough

Reply:
Yes I agree but I will rework them to be more precise